rust/chains/abacus-ethereum/src/retrying.rs

```diff
@@ -118,17 +120,20 @@ impl JsonRpcClient for RetryingProvider<Http> {
                     last_err = HttpClientError::ReqwestError(e);
                 }
                 Err(HttpClientError::JsonRpcError(e)) => {
-                    // This is a client error so we do not want to retry on it.
-                    warn!(error = %e, "JsonRpcError in retrying provider; not retrying.");
-                    return Err(RetryingProviderError::JsonRpcClientError(
-                        HttpClientError::JsonRpcError(e),
-                    ));
+                    if METHODS_TO_NOT_RETRY.contains(&method) {
+                        // This is a client error so we do not want to retry on it.
```

Comment:
It's less clear to me what 'client' error means now that we're selecting from METHODS_TO_NOT_RETRY